packages/shared/src/components/cards/CollectionCard/CollectionCard.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { forwardRef, Ref, useState } from 'react';
+import classNames from 'classnames';
+import { Container, PostCardProps } from '../common';
+import FeedItemContainer from '../FeedItemContainer';
+import { CollectionCardHeader } from './CollectionCardHeader';
+import {
+  getPostClassNames,
+  FreeformCardTitle,
+  CardSpace,
+  CardButton,
+} from '../Card';
+import { WelcomePostCardFooter } from '../WelcomePostCardFooter';
+import ActionButtons from '../ActionButtons';
+import PostMetadata from '../PostMetadata';
+
+export const CollectionCard = forwardRef(function CollectionCard(
+  {
+    children,
+    post,
+    domProps = {},
+    onUpvoteClick,
+    onCommentClick,
+    onMenuClick,
+    onShare,
+    onShareClick,
+    openNewTab,
+    onReadArticleClick,
+    onPostClick,
+  }: PostCardProps,
+  ref: Ref<HTMLElement>,
+) {
+  const [headingFull, toggleHeadingFull] = useState(false);
+  const clamp = (() => {
+    if (post.image) {
+      return 'line-clamp-3';
+    }
+
+    return post.contentHtml ? 'line-clamp-4' : 'line-clamp-9';
+  })();
+
+  return (
+    <FeedItemContainer
+      domProps={{
+        ...domProps,
+        className: getPostClassNames(post, domProps.className, 'min-h-card'),
+        onMouseEnter: () => toggleHeadingFull(true),
+        onMouseLeave: () => toggleHeadingFull(false),
+      }}
+      ref={ref}
+      flagProps={{ pinnedAt: post.pinnedAt }}
+    >
+      <CardButton title={post.title} onClick={() => onPostClick(post)} />
+
+      <CollectionCardHeader
+        post={post}
+        full={headingFull}
+        onMenuClick={onMenuClick}
+      />
+
+      <FreeformCardTitle
+        className={classNames(
+          clamp,
+          'px-2 font-bold !text-theme-label-primary typo-title3',
```

Comment:
Ideally there shouldn't be !important notes if we can avoid them.

Reply:
copy/pasta from the welcome post card 👀  I'll double check if this is actually needed